### PR TITLE
[Feat] Apply the 2FA support to the yakumo's publishing hook

### PR DESCRIPTION
**Pull request by warma16**
Nowadays, I have got some problems in the publishing the Koishi's plugin. Such as the authentication of the npm's. And I'm using the 2FACode to login the npm. But the publishing script cannot figure it out.So I wrote the functions of it.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,5 +24,8 @@
     "mocha": "^9.2.2",
     "shx": "^0.3.4",
     "typescript": "^4.7.4"
+  },
+  "dependencies": {
+    "yakumo": "^0.3.6"
```
Comment (shigma) [inline package.json]: What is diff here for?

Comment (warma16) [reply package.json]: for the vscode's dependency completion😂😂

Comment (warma16) [reply package.json]: > What is diff here for?

 Can this pull request be merged?

Comment (shigma) [reply package.json]: It's generally ok, but I need to make some adjustments before merging.

Comment (warma16) [reply package.json]: OK,I got it

Comment (shigma) [inline package.json]:
```suggestion
```

File: packages/publish/src/index.ts [hunk 2]
```diff
@@ -36,15 +36,22 @@ function getPublishCommand(manager: Manager) {
   return ['yarn', 'npm']
 }
 
-function publish(manager: Manager, path: string, name: string, version: string, tag: string, access: string) {
+function publish(manager: Manager, path: string, name: string, version: string, tag: string, access: string,faCode:string="") {
```
Comment (shigma) [inline packages/publish/src/index.ts]:
```suggestion
function publish(manager: Manager, path: string, name: string, version: string, tag: string, access: string, otp = '') {
```

File: packages/publish/src/index.ts [hunk 3]
```diff
@@ -36,15 +36,22 @@ function getPublishCommand(manager: Manager) {
...
   console.log(`publishing ${name}@${version} ...`)
-  return spawnAsync([
+  let spawnAsyncConfig=[
```
Comment (shigma) [inline packages/publish/src/index.ts]:
```suggestion
  const args = [
```

File: packages/publish/src/index.ts [hunk 4]
```diff
@@ -36,15 +36,22 @@ function getPublishCommand(manager: Manager) {
...
     '--color',
-  ], {
+  ]
+  //logic in detecting isUsing2FAMode
+  let isUsing2FAMode=faCode==""?false:true
+  if(isUsing2FAMode){
+    spawnAsyncConfig.push('--otp')
+    spawnAsyncConfig.push(faCode)
+  }
+  return spawnAsync(spawnAsyncConfig, {
```
Comment (shigma) [inline packages/publish/src/index.ts]:
```suggestion
  if (otp) args.push('--otp', otp)
  return spawnAsync(args, {
```

File: packages/publish/src/index.ts [hunk 5]
```diff
@@ -86,7 +93,8 @@ register('publish', async (project) => {
   for (const path in targets) {
     const { name, version } = targets[path]
     await project.emit('publish.before', path, targets[path])
-    await publish(project.manager, path, name, version, argv.tag ?? (isNext(version) ? 'next' : 'latest'), argv.access ?? 'public')
+    //detecting the argument which is caused the enter 2FACodeMode
+    await publish(project.manager, path, name, version, argv.tag ?? (isNext(version) ? 'next' : 'latest'), argv.access ?? 'public',argv.otp==undefined?"":argv.otp)
```
Comment (shigma) [inline packages/publish/src/index.ts]:
```suggestion
    await publish(project.manager, path, name, version, argv.tag ?? (isNext(version) ? 'next' : 'latest'), argv.access ?? 'public', argv.otp)
```

**Pull request merged**